src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs

```diff
@@ -1681,189 +1611,137 @@ protected override void ProcessRecord()
-                            else if (Status.Equals("Stopped", StringComparison.CurrentCultureIgnoreCase))
-                            {
-                                if (!service.Status.Equals(ServiceControllerStatus.Stopped))
+
+                                ServiceController[] servicedependedon = service.ServicesDependedOn;
+
+                                if ((servicedependedon != null) && (servicedependedon.Length > 0))
                                 {
-                                    //check for the dependent services as set-service dont have force parameter
-                                    ServiceController[] dependentServices = service.DependentServices;
-
-                                    if ((dependentServices != null) && (dependentServices.Length > 0))
-                                    {
-                                        WriteNonTerminatingError(service, null, "ServiceHasDependentServicesNoForce", ServiceResources.ServiceHasDependentServicesNoForce, ErrorCategory.InvalidOperation);
-                                        continue;
-                                    }
-
-                                    ServiceController[] servicedependedon = service.ServicesDependedOn;
-
-                                    if ((servicedependedon != null) && (servicedependedon.Length > 0))
-                                    {
-                                        WriteNonTerminatingError(service, null, "ServiceIsDependentOnNoForce", ServiceResources.ServiceIsDependentOnNoForce, ErrorCategory.InvalidOperation);
-                                        continue;
-                                    }
-                                    //stop service,give the force parameter always true as we have already checked for the dependent services
-                                    //Specify NoWait parameter as always false since we are not adding this switch to this cmdlet
-                                    DoStopService(service, true, true);
+                                    WriteNonTerminatingError(service, null, "ServiceIsDependentOnNoForce", ServiceResources.ServiceIsDependentOnNoForce, ErrorCategory.InvalidOperation);
+                                    return;
                                 }
-                            }
-                            else if (Status.Equals("Paused", StringComparison.CurrentCultureIgnoreCase))
-                            {
-                                if (!service.Status.Equals(ServiceControllerStatus.Paused))
-                                    //pause service
-                                    DoPauseService(service);
+                                //stop service,give the force parameter always true as we have already checked for the dependent services
+                                //Specify NoWait parameter as always false since we are not adding this switch to this cmdlet
+                                DoStopService(service, true, true);
```

Comment:
Since we are touching the code here, can we use named parameters?

Reply:
Will do.